libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/route/ReplayPolylineDecodeStream.kt

```diff
@@ -94,4 +94,21 @@ class ReplayPolylineDecodeStream(
         }
         return points
     }
+
+    /**
+     * Skip the next [minDistance] of the geometry. Similar to decode except that this does not
+     * hold any points in memory.
+     */
+    fun skip(
+        minDistance: Double,
+        @TurfConstants.TurfUnitCriteria units: String = TurfConstants.UNIT_KILOMETERS
+    ) {
+        var travelled = 0.0
+        current?.let { if (minDistance > 0.0 && hasNext()) next() }
+        while (travelled < minDistance && hasNext()) {
+            val previous = current
+            val next = next()
+            travelled += previous?.let { TurfMeasurement.distance(previous, next, units) } ?: 0.0
+        }
+    }
 }
```

Comment:
There is a potential issue with `skip(index)` and that is it depends on the geometry precision.

The current implementation of `skip(distance)` does not do this, but in order to make this function more accurate we would want to inject a new point along the geometry. 

Only sharing that this was considered, but that we're not doing it because there is an assumption that the geometry precision is at a certain level.

Reply:
Hmm while testing this is not behaving well for route alternative switching 🤔

Reply:
Yeah, the precision will matter for more accurate alternative route switching. But that issue can be seen in both approaches at the moment. 

When switching routes the driver also stops, which is an unrelated issue. Considering other improvements out of scope here because they are complicated (can't solve everything at once)

Resolving this

Reply:
Oh my `skip` function was off by an index, my bad. Covered it with tests. The driver still slows down but at least does not move backwards.